light-client/src/supervisor.rs

```diff
@@ -154,6 +157,23 @@ impl Supervisor {
     /// Verify either to the latest block (if `height == None`) or to a given block (if `height == Some(height)`).
     #[pre(self.peers.primary().is_some())]
     fn verify(&mut self, height: Option<Height>) -> VerificationResult {
+        self.verify_inner(height).map(|trusted_lightblock| {
+            // We can now trust this block
+            self.peers
+                .primary_mut()
+                .unwrap() // SAFETY:`verify_inner` wouldn't have returned `Ok` if there were no more primary
+                .trust_block(&trusted_lightblock);
+
+            trusted_lightblock
+        })
+    }
+
+    /// Helper for `verify` to work around limitations of the borrow checker.
```

Comment:
Isn't it possible to hide this by making it a pure (private) function?

Reply:
I guess we could declare it as a function within `verify` itself. Or did you mean as a plain function outside of the supervisor impl?

Reply:
either way (nor sure what works best)